### ISSUE TITLE
Loading dialog continuously open and closes while loading world

### ISSUE THREAD

**Issue by stefaniapedrazzi** [cyberbotics/webots#5493]
On Windows, it happens that when opening a world the loading dialog continuously disappears and re-appears when the following text is displayed "Open World File".
This blinking effect is a little bit disturbing and should be avoided.

I could reproduce this on Windows (didn't check on Ubuntu).

**Comment by omichel** [comment]
I confirm this problem also shows up on macOS.

**Comment by ad-daniel** [comment]
This might actually be a regression, it doesn't happen with the release of R2022b

**Comment by ad-daniel** [comment]
The problem appears to stem from 2 conflicting progress counters:
- [This](https://github.com/cyberbotics/webots/blob/48444fa1f9d6213ef6efc0f69f63dbbd13db692f/src/webots/app/WbApplication.cpp#L53-L54) check says the progress is 100 and therefore the dialog is closed
- meanwhile, occasionally, [this](https://github.com/cyberbotics/webots/blob/48444fa1f9d6213ef6efc0f69f63dbbd13db692f/src/webots/app/WbApplication.cpp#L142-L147) setter re-creates the dialog because according to the parameter it receives, the progress is not 100

basically the two fight with each other

**Comment by omichel** [comment]
I believe we can close this issue now.
